oauthenticator/gitlab.py

```diff
@@ -167,13 +173,24 @@ class GitLabOAuthenticator(OAuthenticator):
             self.log.warning("%s not in group or project whitelist", username)
             return None
 
+    async def _get_gitlab_version(self, access_token):
+        url = '%s/version' % GITLAB_API
+        req = HTTPRequest(url,
+                          method="GET",
+                          headers=_api_headers(access_token),
+                          validate_cert=self.validate_server_cert)
+        resp = await AsyncHTTPClient().fetch(req, raise_error=True)
+        resp_json = json.loads(resp.body.decode('utf8', 'replace'))
+        version_string = re.sub(r'-pre$', '', resp_json['version'])
+        return list(map(int, str.split(version_string, '.')))
```

Comment:
What do you think about skipping the use of a regular expression here, and instead do something like this?

I know that my gitlab version returned was `12.2.5-ee`, so it makes sense to not lock into specifically stripping `-pre` or similar but instead truncating at anything after `-` that separates the `major.minor.patch` part of the string from the other details.

```python
        version_strings = resp_json["version"].split("-")[0].split(".")[:3]
        version_ints = list(map(int, version_strings))
        return version_ints
```

Reply:
Great catch!

Reply:
Fixed